src/masternode.cpp

```diff
@@ -810,8 +810,10 @@ bool CMasternodePing::CheckAndUpdate(CMasternode* pmn, bool fFromNewBroadcast, i
         mnodeman.mapSeenMasternodeBroadcast[hash].second.lastPing = *this;
     }
 
-    pmn->Check(true); // force update, ignoring cache
-    if (!pmn->IsEnabled()) return false;
+    // force update, ignoring cache
+    pmn->Check(true);
+    // relay ping for nodes in ENABLED/EXPIRED/WATCHDOG_EXPIRED state only, skip everyone else
+    if (!pmn->IsEnabled() && !pmn->IsExpired() && !pnm->IsWatchdogExpired()) return false;
```

Comment:
Syntax error: pnm instead of pmn in third position (`!pnm->IsWatchdogExpired()`)

Reply:
yep, fixed it just a few seconds ago in ce0b9af28339f7ad3b6ac0d07dfb81312d14d1c4 :)